### ISSUE TITLE
Docker compile failing for non-root user

### ISSUE THREAD

**Issue by leo-bart** [mbsim-env/build#1]
I cloned the mbsim-env/build repository to my newly created /home/myuser/mbsim folder. Also allowed Docker to run as non-root.

When I run `build.sh` on /home/myuser/mbsim, `aclocal` fails and Docker build report displays the following:

![image](https://user-images.githubusercontent.com/18699508/80604625-59409e80-8a08-11ea-9a9e-10b5a5ef5644.png)

When clicking on the `aclocal failed` link, the following message appears:
```
RUNNING aclocal

autom4te: cannot create autom4te.cache: No such file or directory
aclocal: error: echo failed with exit status: 1
```

**Comment by friedrichatgc** [comment]
I cannot reproduce this issue.
Is /home/myuser/mbsim/build up to date with GitHub? I guess so (you cloned it into a new dir).
Is the docker image mbsimenv/build update to date with DockerHub (running "docker pull mbsimenv/build" should state that the image is update to date)? build.sh does NOT pull new images from DockerHub automatically. So if you have pulled this image already some time ago it may already exists on your machine but is outdated.
Please check this and if it does not help please provide some more informations: OS version, docker version, mbsim-env/build git commit id, mbsimenv/build docker image sha256, ...

**Comment by leo-bart** [comment]
All images are up to date.

I managed to compile everything by uninstalling/reinstalling docker and cloning mbsim-env to a new folder. Then I ran docker as my root user. For some reason, after my second installation I couldn't make docker run as myuser at all!

Some info:
OpenSUSE Leap 15.1
docker 19.03.8
mbsim-env build commit 0c1e74cbf95d12b180bf220ed36ccb49c9de1956

I've never used docker before, so I may be doing somethings wrong.

**Comment by friedrichatgc** [comment]
On most distribution docker can only be run as root after installing. To allow docker for other users some manual steps must be done, see e.g.
https://docs.docker.com/engine/install/linux-postinstall/

I have no idea what was wrong with your first try. But if I understand you right is working now on your site. Please note that you may (I think must) redo everything in a new plain directory if you enable docker to run as a normal user and execute build.sh as a normal user. (another approach would be to change the ownership of all files in the old directory from root to the normal user; but I recommend to redo it from scratch).

**Comment by leo-bart** [comment]
I'm not sure my second try was too successful as well. Docker created some 40 GB of hidden files on my computer (not sure if this should happen).

I did the post-installation procedure indicated on the Docker website when I first installed it. For some reason it is not working and I can only access the Docker daemon using sudo. But this, of course, has nothing to do with mbsim and is probably something with my system.

**Comment by friedrichatgc** [comment]
The "hidden files" from docker should be just about 3GB for the mbsimenv/build image (see "docker image ls"), all docker disc space can be viewed by "docker system df". The hidden dir ".ccache" in your mbsim dir may also contain several GB. The other dir in your mbsim dir may contain also several GB but not 40GB in sum.

All in all this seems to be more docker usage problems, hence closing this issue.

**Comment by leo-bart** [comment]
For the record: I found out why I was not being able to run Docker as non root.

When I installed Docker for the first time, using the [official instructions](https://docs.docker.com/engine/install/), I also allowed rootless access to Docker daemon, which is a procedure that is **not required** to use Docker as non-root user. One of the things I had to do to activate the Docker daemon to any user was to define the `DOCKER_HOST` environment variable using my user's `gid`. Later on I recreated the `docker` user group, which is required to use Docker as a non-root user. The problem was, that when a recreated this `docker` group, its `gid` was changed, but I did not reflect this change on my previously defined `DOCKER_HOST` variable. My system, then, was trying to access the Docker socket using a group user id that was no more assigned to the correct group.

The bottom line is: do not create the `DOCKER_HOST` env variable if you just want to use Docker to build your images. And, in case you need to create it, always check if your `gid`is up to date.
